### PR TITLE
feat: [M3-7579] - Design Tokens (CDS 2.0)

### PR DESCRIPTION
**Pull request by jaalah-akamai**
## Description 📝
Initial pass to replace the hard-coded styles in the `light.ts` theme file with their corresponding design tokens, so that we can achieve consistency in styling and facilitate future theming. We will be doing more structural changes to the theme files in the near future. These design tokens are in sync with **Core Design System (CDS) 2.0** decisions being made for Akamai.

**Benefits:**
- Single source of truth for design decisions
- Autocompletion 
- Type Checking / Inference 
- Design consistency across Akamai projects

## Changes  🔄
- New dependency `"@linode/design-language-system": "^1.0.0"`
    - https://github.com/linode/design-language-system 
- Updated some of our status colors in the app to use the equivalent design token
- Storybook color palettes have been updated automatically.

## Preview 📷
| Before  | After   |
| ------- | ------- |
| Screenshot 2024-06-04 at 5 34 24 PM | Screenshot 2024-06-04 at 5 33 24 PM |
| Screenshot 2024-06-04 at 5 30 44 PM| Screenshot 2024-06-04 at 5 30 30 PM|

### Prerequisites
- Checkout branch
- `yarn install && yarn build && yarn dev`

### Reproduction steps
- Go to http://localhost:3000

### Verification steps 
- Observe differences compared to https://cloud.linode.com/

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [x] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [x] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/features/Users/UserPermissionsEntitySection.tsx [hunk 1]
```diff
@@ -6,45 +6,47 @@
  * I'll create a tech debt ticket in jira to keep track of this issue.
```
Comment (jaalah-akamai) [inline packages/manager/src/features/Users/UserPermissionsEntitySection.tsx:6]: This entire file was untouched - I just ran ESLINT against it.

File: packages/manager/src/components/PrimaryNav/PrimaryNav.styles.ts [hunk 2]
```diff
@@ -75,8 +75,8 @@ const useStyles = makeStyles<void, 'linkItem'>()(
           opacity: 1,
         },
         '& svg': {
-          color: theme.color.teal,
-          fill: theme.color.teal,
```
Comment (jaalah-akamai) [inline packages/manager/src/components/PrimaryNav/PrimaryNav.styles.ts:79]: Teal was green in production, whereas it's actually teal with new design tokens. I just updated this to use out new status color for success (green).

**Review by bnussman-akamai — COMMENTED** [review]
Neat! It's gonna take me a bit to get use to all of the new colors, but things look solid!

The table header looks much different, and slightly less appealing now in my opinion. Is this change to bring us into alignment with Akamai's tables?

**Comment by jaalah-akamai** [comment]
> The table header looks much different, and slightly less appealing now in my opinion. Is this change to bring us into alignment with Akamai's tables?

Yes! The table will look much different once we actually apply the Table tokens. We can either decide to make that change now or in an upcoming PR. I was hesitant in doing anything drastic with this first PR and was more trying to do a 1:1 mapping.

File: packages/manager/src/foundations/themes/light.ts [hunk 3]
```diff
@@ -1436,30 +1438,30 @@ export const lightTheme: ThemeOptions = {
...
     },
     mode: 'light',
     primary: primaryColors,
     secondary: primaryColors,
     success: {
-      dark: '#00b159',
-      light: '#00b159',
-      main: '#00b159',
+      dark: Color.Green[70],
+      light: Color.Green[10],
```
Comment (bnussman-akamai) [inline packages/manager/src/foundations/themes/light.ts:1455]: I'm not sure exactly why MUI has a `light`, `dark`, and `main`, but I think this is causing our green "success" color to look more white than green. I've "solved" this in the past by just making `light`, `dark`, and `main` all the same value. I'm not sure exactly what should be done here

| Before | After |
|--------|-------|
|   Screenshot 2024-01-02 at 2 20 41 PM  |     Screenshot 2024-01-02 at 2 17 45 PM  |

Comment (bnussman-akamai) [reply packages/manager/src/foundations/themes/light.ts:1455]: Ohhh. I think MUI picks `light` if `palette.mode` is `light` and  `dark`  if palette.mode is `dark`. That's why light mode uses `Color.Green[10]`

Comment (jaalah-akamai) [reply packages/manager/src/foundations/themes/light.ts:1457]: Ok so I fixed the issue, but I'm open to how we interpret `dark`, `light` and `main` for our needs. I was trying to follow our status color shades with `dark` being the main (to use the same terminology) fill color... however MUI describes it a little different https://mui.com/material-ui/customization/palette/.
 
Screenshot 2024-01-02 at 9 12 05 PM

cc: @abailly-akamai

**Review by bnussman-akamai — APPROVED** [review]
Not seeing any breaking changes! This will take getting use to! 🎨

For our Button, the dark mode hover state is much less obvious than before. Do we need to change the colors so that there is more of a difference when you hover in dark mode?

https://github.com/linode/manager/assets/115251059/3e552a0b-607f-4b75-b13a-de02ea4851ca

File: packages/manager/src/foundations/themes/light.ts [hunk 4]
```diff
@@ -546,7 +550,7 @@ export const lightTheme: ThemeOptions = {
     MuiDialog: {
       styleOverrides: {
         paper: {
-          boxShadow: '0 0 5px #bbb',
+          boxShadow: `0 0 5px ${Color.Neutrals[50]}`, // TODO: This was the closest color according to our palette
```
Comment (abailly-akamai) [inline packages/manager/src/foundations/themes/light.ts:553]: Should those use elevation tokens?

File: packages/manager/src/foundations/themes/light.ts [hunk 5]
```diff
@@ -928,7 +932,7 @@ export const lightTheme: ThemeOptions = {
     MuiPaper: {
       styleOverrides: {
         outlined: {
-          border: '1px solid #e7e7e7',
+          border: `1px solid ${Color.Neutrals[30]}`,
```
Comment (abailly-akamai) [inline packages/manager/src/foundations/themes/light.ts:935]: Do we have border tokens?

File: packages/manager/src/foundations/themes/light.ts [hunk 6]
```diff
@@ -1243,7 +1247,7 @@ export const lightTheme: ThemeOptions = {
             color: textColors.tableHeader,
           },
           '&:focus': {
-            outline: '1px dotted #999',
+            outline: `1px dotted ${Color.Neutrals[60]}`, // TODO: This was the closest color according to our palette
```
Comment (abailly-akamai) [inline packages/manager/src/foundations/themes/light.ts:1250]: same about border tokens

File: packages/manager/src/foundations/themes/light.ts (hunk 4 above)
Comment (jaalah-akamai) [reply packages/manager/src/foundations/themes/light.ts:553]: I'm going to leave this one untouched for now - we can revisit when we look at the Dialog component.

**Review by abailly-akamai — APPROVED** [review]
Code and UI look good. Did not catch anything else unexpected.

I really like this update and the improved contrast on form items. The table header is a bit hard to get used to (it will get even darker!) but positive change AFAIAC!

**Comment by abailly-akamai** [comment]
@jaalah-akamai also you'll need a rebase

**Comment by jaalah-akamai** [comment]
@abailly-akamai Reverted borders on several components until we can review them later and fixed the marketplace reference since our brand blues have changed. It's looking good now 👍 

Screenshot 2024-01-17 at 9 38 19 AM

**Review by bnussman-akamai — COMMENTED** [review]
The only drawback of this PR I've noticed is that we'll lose type safety on the colors. This isn't a deal breaker, but I always found it super useful to be able to see what color was used under the hood


| Before  | This PR   |
|---|---|
| Screenshot 2024-01-24 at 10 44 35 AM  |  Screenshot 2024-01-24 at 10 44 17 AM |

I guess to fix this, we'd need to export `as const` from the design systems package - if that is even possible

**Comment by abailly-akamai** [comment]
@jaalah-akamai do we want to wait for the next version of https://github.com/linode/design-language-system and update this PR with the latest lib improvement?

File: packages/manager/src/foundations/themes/dark.ts [hunk 7]
```diff
@@ -358,41 +459,61 @@ export const darkTheme: ThemeOptions = {
...
+            color: Select.Hover.Text,
           },
-          backgroundColor: '#444',
-          border: '1px solid #222',
-          color: primaryColors.text,
+          backgroundColor: Select.Default.Background,
+          borderColor: Select.Default.Border,
+          // TODO: designTokens - figure out why:
+          // - Select.Default.Text and Select.Focus.Text would be different colors
+          // - Why Select.Default.Text (#a3a3ab) looks like a disabled color (maybe that's just the discrepancy)
```
Comment (abailly-akamai) [inline packages/manager/src/foundations/themes/dark.ts:505]: @jaalah-akamai any idea about this one? I changed it for now but using `Select.Default.Text` looks like a disabled style then color changes on focus which is not what we want

**Review by dwiley-akamai — APPROVED** [review]
Visually audited the app in light and dark mode ✅

Nice work, I like some aspects of this better than the current design, such as the blue hover/selection state for primary nav items instead of green.

<details>
<summary>Screenshots & Questions</summary>

Screenshot 2024-06-20 at 12 44 13 PM

1. Do we expect to keep the jet black background? It feels a bit jarring. And it looks like we have some stray miscoloration in the bottom left.
2. Do we expect to keep the light gray borders/outlines on the elements? It makes the app look busier than the current dark gray borders/outlines IMO.

Screenshot 2024-06-20 at 12 36 15 PM
</details>

**Review by bnussman-akamai — APPROVED** [review]
List of things I've noticed. Looks great!

> [!note]
> I might keep adding stuff to this comment so keep an eye out for new items added

<details>
  <summary>Popovers/Menus have no contrast</summary>
  I feel like popovers/menus should have a different background color than the app's background color.
  <br />
  
</details>

<details>
  <summary>Top Nav icon colors inconsistant</summary>
 Can we make the hamburger icon the same color as the other ones?
  <br />

  Screenshot 2024-06-20 at 3 25 32 PM
</details>

<details>
  <summary>Random black in the bottom left on smaller viewports</summary>
  Can this be fixed?
  <br />

  
Screenshot 2024-06-20 at 3 27 34 PM

</details>

<details>
  <summary>Can we add a contrasting background color to the footer?</summary>
  Previously, we only had contrast in dark mode. I'd personally like to see the footer have a contrasting background color it in both themes
  <br />

  
Screenshot 2024-06-20 at 3 28 59 PM


</details>


<details>
  <summary>Can we make entity icons not green?</summary>
  I'm not sure if these were intentionally kept green. Can we make then blue, gray, or white?
  <br />

  
Screenshot 2024-06-20 at 3 33 10 PM



</details>

**Comment by abailly-akamai** [comment]
Addressed some of the feedback from @dwiley-akamai & @bnussman-akamai which I will list here (dark theme only, did not touch light theme):

> Do we expect to keep the jet black background? It feels a bit jarring. And it looks like we have some stray miscoloration in the bottom left.

- miscoloration (also mentioned by Banks) has been fixed 
- i had to re-work the jet-black/primary nav/top nav handling cause styling was a bit broken - **eventually** the jet black will be our top bar bg (CDS styles) but we will have to rework how the primary nav is built to make them work together. for now i lightened it some to transition over smoothly to the future CDS styles and kept the sidebar somewhat similar to what it is now (darker) to minimize impact and keep experience nice

> Do we expect to keep the light gray borders/outlines on the elements? It makes the app look busier than the current dark gray borders/outlines IMO.

yes, these are the styles for CDS 2.0 which Cloud Manager is adopting now and will increase adoption in the next major update

> Popovers/Menus have no contrast

yeah i agree this isn't the best experience so I added contrast back

> Top Nav icon colors inconsistent

fixed!

> Can we add a contrasting background color to the footer?

leaving this one out cause the light theme does not have it and i don't think we need to in the dark mode either (trying to keep themes consistent in terms of main elements)

> Can we make entity icons not green?

That will be coming in the future but for now marketing/product styling is out of scope.

**Pull request merged**